controllers/tf_controller.go

```diff
@@ -232,6 +232,21 @@ func (r *TerraformReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	// check dependencies, if not being deleted
 	if len(terraform.Spec.DependsOn) > 0 && !isBeingDeleted(terraform) {
 		if err := r.checkDependencies(sourceObj, terraform); err != nil {
+			if acl.IsAccessDenied(err) {
+				traceLog.Info("The cross-namespace dependency was denied by reconciler.NoCrossNamespaceRefs")
+
+				terraform = infrav1.TerraformNotReady(terraform, sourceObj.GetArtifact().Revision, infrav1.ArtifactFailedReason, err.Error())
```

Comment:
Here too -- ArtifactFailed seems like the wrong reason (see comment in test case).